main.tf

```diff
@@ -152,5 +152,10 @@ resource "aws_iam_role_policy_attachment" "kms_policy_attachment" {
   count      = var.kms_key_arn != "" ? 1 : 0
   role       = module.lambda.iam_role_name
   policy_arn = aws_iam_policy.kms_policy[count.index].arn
-} 
+}
 
+resource "aws_lambda_function_url" "lambda_url" {
+  count              = var.enable && lookup(var.architecture, "function_url", false) ? 1 : 0
+  function_name      = module.lambda.arn
+  authorization_type = "AWS_IAM"
```

Comment:
Let's swap this to a variable that validates whether either `AWS_IAM` or `NONE`,  default to `AWS_IAM` as the default (more secure) option out of the gate. Great points during Zoom PR 🏅 .